base/libgit2/types.jl

```diff
@@ -177,22 +177,44 @@ function RemoteCallbacks{P<:AbstractPayload}(credentials::Ptr{Void}, payload::Nu
     end
 end
 
-immutable FetchOptions
-    version::Cuint
-    callbacks::RemoteCallbacks
-    prune::Cint
-    update_fetchhead::Cint
-    download_tags::Cint
-end
-FetchOptions(; callbacks::RemoteCallbacks = RemoteCallbacks(),
-               prune::Cint = Consts.FETCH_PRUNE_UNSPECIFIED,
-               update_fetchhead::Cint = one(Cint),
-               download_tags::Cint = Consts.REMOTE_DOWNLOAD_TAGS_AUTO
-) = FetchOptions(one(Cuint),
-                 callbacks,
-                 prune,
-                 update_fetchhead,
-                 download_tags)
+if LibGit2.version() >= v"0.24.0"
```

Comment:
Same here. New field is just appended, and old field are in their original places. This appended field would not break any library call even if the library is not of version v0.24.